prboom2/src/m_menu.c

```diff
@@ -5823,6 +5828,31 @@ void M_StartControlPanel (void)
   itemOn = currentMenu->lastOn;   // JDC
 }
 
+
+/////////////////////////////////////////////////////////////////////////////
+//
+// Menu Shaded Overlay Stuff
+//
+// This displays a dark overlay under certain screens of the menus
+
+int screenshade = 20;
+
+dboolean fadeBG(void)
+{
+  return dsda_IntConfig(dsda_config_menu_background) == 1;
+}
+
+dboolean M_MenuIsShaded(void)
+{
+  int Options = (setup_active || currentMenu == &OptionsDef || currentMenu == &SoundDef);
+  return fadeBG() && Options;
+}
+
+static void M_ShadedScreen(int scrn)
+{
+  V_DrawShaded(scrn, 0, 0, SCREENWIDTH, SCREENHEIGHT, screenshade);
```

Comment:
In general, it is good practice to reduce the number of global variables as much as possible. Doom is not good at this, but let's not make it worse. Let's remove the global variable `screenshade`, but keep the function parameter `screenshade`, then the difference with Nyan Doom would be minimal:
```diff
diff --git a/prboom2/src/am_map.c b/prboom2/src/am_map.c
index 3650a72c9..eeda64153 100644
--- a/prboom2/src/am_map.c
+++ b/prboom2/src/am_map.c
@@ -2801,7 +2801,7 @@ void AM_Drawer (dboolean minimap)
   if (!automap_overlay) // cph - If not overlay mode, clear background for the automap
     V_FillRect(FB, f_x, f_y, f_w, f_h, (byte)mapcolor_p->back); //jff 1/5/98 background default color
   if (automap_overlay == 2 && !M_MenuIsShaded())
-    V_DrawShaded(FB, f_x, f_y, f_w, f_h, screenshade);
+    V_DrawShaded(FB, f_x, f_y, f_w, f_h, SCREENSHADE);
 
   if (map_textured)
   {
diff --git a/prboom2/src/m_menu.c b/prboom2/src/m_menu.c
index ce58856f3..0cbccddf4 100644
--- a/prboom2/src/m_menu.c
+++ b/prboom2/src/m_menu.c
@@ -5835,8 +5835,6 @@ void M_StartControlPanel (void)
 //
 // This displays a dark overlay under certain screens of the menus
 
-int screenshade = 20;
-
 dboolean fadeBG(void)
 {
   return dsda_IntConfig(dsda_config_menu_background) == 1;
@@ -5850,7 +5848,7 @@ dboolean M_MenuIsShaded(void)
 
 static void M_ShadedScreen(int scrn)
 {
-  V_DrawShaded(scrn, 0, 0, SCREENWIDTH, SCREENHEIGHT, screenshade);
+  V_DrawShaded(scrn, 0, 0, SCREENWIDTH, SCREENHEIGHT, SCREENSHADE);
 }
 
 //
diff --git a/prboom2/src/m_menu.h b/prboom2/src/m_menu.h
index 2048b4000..a69bc267c 100644
--- a/prboom2/src/m_menu.h
+++ b/prboom2/src/m_menu.h
@@ -51,7 +51,7 @@ dboolean M_Responder (event_t *ev);
 
 dboolean fadeBG(void);
 dboolean M_MenuIsShaded(void);
-extern int screenshade;
+#define SCREENSHADE 20
 
 // Called by main loop,
 // only used for menu (skull cursor) animation.
```

Reply:
So in Nyan Doom, `screenshade` is a variable that changes due to some gradual fade logic. To keep this consistent between both DSDA and Nyan, I think I'll replace `int screenshade` here with a `#define FULLSHADE 20` which Nyan uses to compare `screenshade` to.